tracing-subscriber/src/layer/mod.rs

```diff
@@ -0,0 +1,1183 @@
+//! the `http_access` target, while logging other spans and events to
+//! standard out, a [`Filter`] can be added to the access log layer:
+//!
+//! ```
+//! use tracing_subscriber::{filter, prelude::*};
+//!
+//! // Generates an HTTP access log.
+//! let access_log = // ...
+//!     # filter::LevelFilter::INFO;
+//!
+//! // Add a filter to the access log layer so that it only observes
+//! // spans and events with the `http_access` target.
+//! let access_log = access_log.with_filter(filter::filter_fn(|metadata| {
+//!     // Returns `true` if and only if the span or event's target is
+//!     // "http_access".
+//!     metadata.target() == "http_access"
+//! }));
+//!
+//! // A general-purpose logging layer.
+//! let fmt_layer = tracing_subscriber::fmt::layer();
+//!
+//! // Build a subscriber that combines the access log and stdout log
+//! // layers.
+//! tracing_subscriber::registry()
+//!     .with(fmt_layer)
+//!     .with(access_log)
+//!     .init();
+//! ```
+//!
+//! Multiple layers can have their own, separate per-layer filters. A span or
+//! event will be recorded if it is enabled by _any_ per-layer filter, but it
+//! will be skipped by the layers whose filters did not enable it. Building on
+//! the previous example:
+//!
+//! ```
+//! use tracing_subscriber::{filter::{filter_fn, LevelFilter}, prelude::*};
+//!
+//! let access_log = // ...
+//!     # LevelFilter::INFO;
+//! let fmt_layer = tracing_subscriber::fmt::layer();
+//!
+//! tracing_subscriber::registry()
+//!     // Add the filter for the "http_access" target to the access
+//!     // log layer, like before.
+//!     .with(access_log.with_filter(filter_fn(|metadata| {
+//!         metadata.target() == "http_access"
+//!     })))
+//!     // Add a filter for spans and events with the INFO level
+//!     // and below to the logging layer.
+//!     .with(fmt_layer.with_filter(LevelFilter::INFO))
+//!     .init();
+//!
+//! // Neither layer will observe this event
+//! tracing::debug!(does_anyone_care = false, "a tree fell in the forest");
+//!
+//! // This event will be observed by the logging layer, but not
+//! // by the access log layer.
+//! tracing::warn!(dose_roentgen = %3.8, "not great, but not terrible");
+//!
+//! // This event will be observed only by the access log layer.
+//! tracing::trace!(target: "http_access", "HTTP request started");
+//!
+//! // Both layers will observe this event.
+//! tracing::error!(target: "http_access", "HTTP request failed with a very bad error!");
+//! ```
+//!
+//! A per-layer filter can be applied to multiple [`Layer`]s at a time, by
+//! combining them into a [`Layered`] layer using [`Layer::and_then`], and then
+//! calling [`Layer::with_filter`] on the resulting [`Layered`] layer.
+//!
+//! For example, if two layers `layer_a` and `layer_b` should only recieve the
+//! [`INFO`] [level] and above, but a third `layer_c` should recieve spans and
+//! events with the [`DEBUG`] [level]:
```

Comment:
Maybe the prose is a little too purple, but there _is_ a typo in "recieve" (should be "recieve")

```suggestion
//! Consider the following:
//! - `layer_a` and `layer_b`, which should only receive spans and events at
//!     [`INFO`] [level] and above.
//! - A third layer, `layer_c`, which should receive spans and events with
//!    the [`DEBUG`] [level].
//! The layers and filters would be composed thusly:
```